source/common/http/http2/codec_impl.h

```diff
@@ -181,25 +181,13 @@ class ConnectionImpl : public virtual Connection,
    * Base class for client and server side streams.
    */
   struct StreamImpl : public virtual StreamEncoder,
-                      public Stream,
                       public LinkedObject<StreamImpl>,
                       public Event::DeferredDeletable,
-                      public StreamCallbackHelper,
+                      public Http::MultiplexedStreamImplBase,
                       public ScopeTrackedObject {
 
     StreamImpl(ConnectionImpl& parent, uint32_t buffer_limit);
-    ~StreamImpl() override;
-    // TODO(mattklein123): Optimally this would be done in the destructor but there are currently
-    // deferred delete lifetime issues that need sorting out if the destructor of the stream is
-    // going to be able to refer to the parent connection.
-    virtual void destroy();
-    void disarmStreamIdleTimer() {
-      if (stream_idle_timer_ != nullptr) {
-        // To ease testing and the destructor assertion.
-        stream_idle_timer_->disableTimer();
-        stream_idle_timer_.reset();
-      }
-    }
+    void destroy() override;
```

Comment:
generally for overrides we comment the class it's from

// Http::MultiplexedStreamImplBase